### PR TITLE
Move simplification enums to Qgis, promote to enum class, and remove duplicate enum

### PR DESCRIPTION
**Pull request by nyalldawson**

### CODE REVIEW

File: src/core/qgis.h [hunk 1]
```diff
@@ -3894,7 +3936,8 @@ class CORE_EXPORT Qgis
     /**
      * Legend JSON export flags.
      *
-     * Flags to control JSON attributes when exporting a legend in JSON format.
+     * Flags to co
+    *    Q_DECLARE_OPERATORS_FOR_FLAGS( QgsVectorSimplifyMethod::SimplifyHints )ntrol JSON attributes when exporting a legend in JSON format.
```
Comment (rouault) [inline src/core/qgis.h:3940]: accidental paste:
```suggestion
     * Flags to control JSON attributes when exporting a legend in JSON format.
```

Comment (nyalldawson) [reply src/core/qgis.h]: Oops, silly middle-button paste

**Comment by nyalldawson** [comment]
I've also done some raster ones, in response to https://github.com/qgis/QGIS/pull/57663

File: src/core/raster/qgsrasterinterface.h [hunk 2]
```diff
@@ -227,16 +212,20 @@ class CORE_EXPORT QgsRasterInterface
...
-      return QgsRasterInterface::NoCapabilities;
-    }
+    // TODO QGIS 4.0 -- rename to interfaceCapabilities, to avoid confusion with QgsRasterDataProvider::providerCapabilities
+    // (which inherits this class)
 
     /**
-     *  Returns the raster interface capabilities in friendly format.
+     * Returns the capabilities supported by the interface.
+     */
+    virtual Qgis::RasterInterfaceCapabilities capabilities() const;
```
Comment (nirvn) [inline src/core/raster/qgsrasterinterface.h:221]: I'd also suggest for QGIS 4.0 to rename this to interfaceCapabilities() to make it clearer. I find the current providerCapabilities() + capabilities() pair of functions prone to confusion.

**Review by nirvn — APPROVED** [review]

File: src/core/raster/qgsrasterinterface.h (hunk 2 above)
Comment (nyalldawson) [reply src/core/raster/qgsrasterinterface.h:221]: Scroll up a bit, I've already added a TODO 😉

**Pull request merged**